parentchain/src/lib.rs

```diff
@@ -0,0 +1,61 @@
+#![cfg_attr(not(feature = "std"), no_std)]
+
+pub use pallet::*;
+
+#[frame_support::pallet]
+pub mod pallet {
+	use crate::weights::WeightInfo;
+	use frame_support::pallet_prelude::*;
+	use frame_system::pallet_prelude::*;
+
+	#[pallet::pallet]
+	#[pallet::generate_store(pub(super) trait Store)]
+	pub struct Pallet<T>(_);
+
+	/// Configuration trait.
+	#[pallet::config]
+	pub trait Config: frame_system::Config {
+		type WeightInfo: WeightInfo;
+	}
+
+	/// The current block number being processed. Set by `execute_block`.
+	#[pallet::storage]
+	#[pallet::getter(fn block_number)]
+	pub(super) type Number<T: Config> = StorageValue<_, T::BlockNumber, ValueQuery>;
+
+	/// Hash of the previous block.
+	#[pallet::storage]
+	#[pallet::getter(fn parent_hash)]
+	pub(super) type ParentHash<T: Config> = StorageValue<_, T::Hash, ValueQuery>;
+
+	/// Map of block numbers to block hashes.
+	#[pallet::storage]
+	#[pallet::getter(fn block_hash)]
+	pub(super) type BlockHash<T: Config> = StorageValue<_, T::Hash, ValueQuery>;
+
+	#[pallet::hooks]
+	impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {}
+
+	#[pallet::call]
+	impl<T: Config> Pallet<T> {
+		#[pallet::weight(<T as Config>::WeightInfo::insert_block())]
+		pub fn insert_block(
```

Comment:
As we don't 'insert` something into a map, but rather set a value, I'd go here for: `set_block`.

Also, I wonder if we should change the function signature to take a `Block` or `Header` as an argument (not sure here). I believe I could make sense as we have a more robust interface (not as sensible to code changes).

Reply:
I didn't call it set_block because I got a warning. 

"methods called `set_*` usually take self by mutable reference; consider choosing a less ambiguous name"

Can we find a name that fulfills your request and solves the warning🥴?

Reply:
I only get it in the IDE, do you also get it while compiling?

Reply:
No you are right. Only in the IDE. -> I'll change it. 

About your second question: Bigna said she also thought about it, but did not come to a conclusion yet 🤣
For me it is difficult to discuss about that. I do not have a strong oppinion

Reply:
I was also thinking about this, so lets change it. Sorry @gaudenzkessler, I hope that's okay for you.

 Our current `update_states` function in the worker (https://github.com/integritee-network/worker/blob/17c7624f00ad4cf4b0739f9703d43217bb61d188/enclave-runtime/src/lib.rs#L737-L740), where I'd suggest to place the parentchain_pallet call, takes the header as an argument. If you, @clangenb, also don't see a reason to pass the whole block, we should probably just use the header.

Reply:
Yes, it should be the header.